solarwinds_apm/distro.py

```diff
@@ -67,6 +68,21 @@
 class SolarWindsDistro(BaseDistro):
     """OpenTelemetry Distro for SolarWinds reporting environment"""
 
+    _cnf_dict = None
+    _instrumentor_metrics_enabled = None
+
+    def __new__(cls, *args, **kwargs):
+        # Maintain singleton pattern and cache SW APM user config
+        # for Distro lifecycle at auto-instrumentation
```

Comment:
Implementation detail: the Distro class now caches `_cnf_dict` as contents read from a valid JSON config file once at singleton creation. This is so the distro only reads the file once to calculate if Otel Python metrics enabled for each call to `load_instrumentor`. Auto-instrumentation still reads the file twice (another time by the Configurator to get all other settings); Distro and Configurator can't share one ApmConfig due to lifecycles when [sitecustomize](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/748c92592d2f476199667629defce4a3bca9ecc9/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py#L37-L39) is used.